include/faabric/endpoint/FaabricEndpoint.h

```diff
@@ -38,10 +38,9 @@ class FaabricEndpoint
   public:
     FaabricEndpoint();
 
-    FaabricEndpoint(
-      int port,
-      int threadCount,
-      std::shared_ptr<HttpRequestHandler> requestHandlerIn = nullptr);
+    FaabricEndpoint(int port,
```

Comment:
Before, we allowed users of the `FaabricEndpoint` not to specify a request handler, and we would use `FaabricEndpointHandler` by default.

Unfortunately, this means that any other class implementing a different endpoint (thus using `FaabricEndpoint` but a different endpoint handler than `FaabricEndpointHandler`) still has a link-time dependency to `FaabricEndpointHandler` (i.e. needs to provide all the symbols used by `FaabricEndpointHandler`). In turn, this means that we need to link with `faabric::scheduler` even if not using any symbols therein defined.

The aim of this PR is to split from the scheduler (eventually turning the existing scheduler into a form of local scheduler), so we don't want to take that link-time dependency.